src/main/java/org/mitre/synthea/world/agents/Provider.java

```diff
@@ -176,72 +194,121 @@ private static Provider findService(Person person,
+      
+        servicesProvided.clear();
+        statesLoaded.add(state);
+        statesLoaded.add(Location.getAbbreviation(state));
+        statesLoaded.add(Location.getStateName(state));
+      } catch (IOException e) {
+        System.err.println("ERROR: unable to load providers for state: " + state);
+        e.printStackTrace();
+      }
     }
   }
 
   /**
    * Read the providers from the given resource file, only importing the ones for the given state.
+   * THIS method is for loading providers and generating clinicians with specific specialties
    * 
    * @param state Name of the current state, ex "Massachusetts"
    * @param abbreviation State abbreviation, ex "MA"
    * @param filename Location of the file, relative to src/main/resources
    * @param servicesProvided Set of services provided by these facilities
    * @throws IOException if the file cannot be read
    */
-  public static void loadProviders(String state, String abbreviation, String filename,
+  public static void loadProviders(String state, String abbreviation, String filename, 
       Set<String> servicesProvided)
       throws IOException {
     String resource = Utilities.readResource(filename);
-    List<? extends Map<String,String>> csv = SimpleCSV.parse(resource);
-
-    for (Map<String,String> row : csv) {
+    Iterator<? extends Map<String,String>> csv = SimpleCSV.parseLineByLine(resource);
+    
+    while (csv.hasNext()) {
+      Map<String,String> row = csv.next();
       String currState = row.get("state");
-
+      
       // for now, only allow one state at a time
       if ((state == null)
           || (state != null && state.equalsIgnoreCase(currState))
           || (abbreviation != null && abbreviation.equalsIgnoreCase(currState))) {
+    
         Provider parsed = csvLineToProvider(row);
-
         parsed.servicesProvided.addAll(servicesProvided);
+        
         if ("Yes".equals(row.remove("emergency"))) {
           parsed.servicesProvided.add(Provider.EMERGENCY);
         }
-
+        
         // add any remaining columns we didn't explicitly map to first-class fields
         // into the attributes table
         for (Map.Entry<String, String> e : row.entrySet()) {
           parsed.attributes.put(e.getKey(), e.getValue());
         }
         
+        String city = parsed.city;
+        String address = parsed.address;
+
+        if (row.get("hasSpecialties") == null || row.get("hasSpecialties").equals("FALSE")) {
+          parsed.clinicianMap.put(ClinicianSpecialty.GENERAL_PRACTICE, 
+              parsed.generateClinicianList(1, ClinicianSpecialty.GENERAL_PRACTICE));
+        } else {
+          for (String specialty : ClinicianSpecialty.getSpecialties()) { 
+            if (row.get(specialty) != "") {
+              if (!row.get(specialty).equals("0")) {
+                parsed.clinicianMap.put(specialty, 
+                    parsed.generateClinicianList(Integer.parseInt(row.get(specialty)), specialty));
```

Comment:
While you are looking at the note below can you also clean this section up? Calling `row.get(specialty)` 3 times here means it's time to extract to a variable. Also always compare 2 strings using .equals not ==. Plus the 2 nested `if`s could be merged together. 
Something like:
```
String specialtyCount = row.get(specialty);
if (specialtyCount != null && !specialtyCount.trim().equals("")) {
 clinicianMap.put(....)
}
```